src/http_workers/work_runner.rs

```diff
@@ -94,11 +97,58 @@ where
         }
     }
 
+    async fn upload_once<U: AsRef<str>>(
+        &self,
+        twin: &Twin,
+        uri_path: U,
+        msg: &mut Message,
+        payload: UploadPayload,
+    ) -> Result<(), SendError> {
+        let mut mpart = MultipartRequest::default();
+
+        mpart.add_file("upload", payload.path);
+        let uri = uri_builder(uri_path, &twin.address);
+        log::debug!("sending upload to '{}'", uri);
+
+        let request = Request::post(uri)
+            .header(
+                header::CONTENT_TYPE,
+                format!("multipart/form-data; boundary={}", mpart.get_boundary()),
+            )
+            .header("rmb-upload-cmd", payload.cmd)
+            .header("rmb-source-id", payload.source)
+            .header("rmb-timestamp", payload.timestamp)
+            .header(
+                "rmb-signature",
+                payload.signature.unwrap_or_else(|| "".to_string()),
+            )
+            .body(Body::wrap_stream(mpart))
+            .map_err(|err| anyhow!("{:?}", err))?;
```

Comment:
If you failed to create the request object, that is definitely a Terminal error hence
```suggestion
            .map_err(|err| SendError::Terminal(err.into()))?;
```